src/components/HydrantList.js

```diff
@@ -23,10 +26,53 @@ const styles = theme => ({
 
 class HydrantList extends React.Component {
 
-  state = {
-    open: true,
+
+  constructor(props){
+    super(props)
+    this.state = {
+      open: true,
+      prefix: this.getPrefix(props.trailHydrants),
+    }
   }
 
+  componentWillReceiveProps(nextProps) {
+    const prefix = this.getPrefix(nextProps.trailHydrants)
+    this.setState({
+        prefix
+    })
+
+  }
+
+  getPrefix = (hydrants) => {
+    if (hydrants.size > 1) {
+      return hydrants.reduce((accum, val) => {
+        let name = val.get('name')
+        for (let i = 0; i < accum.length; i++) {
+          if (accum[i] !== name[i]) {
+            return accum.slice(0, i)
+          }
+        };
+        return accum
+      }, hydrants.getIn([0,'name']) )
```

Comment:
The initializer is there so that we start off with the first hydrant's name, otherwise calling accum.get('name') after the first pass would break becuase it wouldn't be an object anymore.